src/box/space.c

```diff
@@ -431,6 +431,70 @@ space_delete(struct space *space)
 	space->vtab->destroy(space);
 }
 
+int
+space_foreach(int (*func)(struct space *sp, void *udata), void *udata)
+{
+	struct space *space = space_by_id(BOX_SPACE_ID);
+	if (space == NULL)
+		return -1;
+	struct index *pk = space_index(space, 0);
+	if (pk == NULL)
+		return -1;
```

Comment:
Neither `space_by_id` nor `space_index` sets diag on failure. You should use `space_cache_find` and `index_find` if you want to fail `space_foreach` if `box.space._space` doesn't exist. However, `box.space._space` must exist, since we insert it into the cache even if there's no such record in the snap file so I think it's better to change this to an assertion.